Fixing support for Python 3.10

mbrea-c opened the pull request:
Due to [this asyncio commit](https://github.com/python/cpython/commit/b9127dd6eedd693cfd716a4444648864e2e00186), initializing ```PulseAsync``` now raises a ```TypeError``` (see [here](https://github.com/python/cpython/commit/b9127dd6eedd693cfd716a4444648864e2e00186#diff-e5b10943fef60b34a59e7006ba041003ed376c5e52d7ba1fa40d323b598f4063R18)).

My fix is to pass the ```loop``` parameter to ```asyncio.Event``` only in older python versions, where it was necessary.

All unit tests passed.

mhthies commented:
Thanks for testing compatibility and investigating in that issue!

What I don't like about your fix is that the `loop` parameter is still present as an init parameter but only has an effect for Python versions up to 3.9. We should at least document that behaviour. In addition, I would explicitly mark the parameter as deprecated and remove it in a future version of this library, as it has never actually been needed:

I just [learned](https://stackoverflow.com/a/60315290/10315508) that the parameter has originally been introduced in the Python standard library to work around the insufficient behaviour of `asyncio.get_current_loop()` up to Python 3.5. However, this library has only suport for Python 3.6 and newer. So we haven't needed the loop parameter for this purpose. I have to admit, that I (mis)used the parameter for special test fixtures in the past, but that has always been a hack.

You could change the GitHub action workflow to test with Python 3.10 instead of 3.9.
(And don't worry about the test failure – it's a bug in the workflow script that I will fix on the `main` branch.)

mbrea-c commented:
> What I don't like about your fix is that the `loop` parameter is still present as an init parameter but only has an effect for Python versions up to 3.9. We should at least document that behaviour. In addition, I would explicitly mark the parameter as deprecated and remove it in a future version of this library, as it has never actually been needed:

Good point. I just added a deprecation warning when the ```loop``` parameter is not ```None```. Is that what you had in mind?

Also added Python 3.10 to the build workflow like you mentioned.

mhthies commented:
I would have replaced the Python 3.9 test task with Python 3.10, but … one task more sholdn't hurt GitHub. ;)

The deprecation warning is exactly what I head in mind. Thanks!
